service/lib/agama/dbus/storage/manager.rb

```diff
@@ -400,7 +390,16 @@ def configure_with_current
           return unless config_json
 
           configure(config_json)
-          self.ProposalChanged
+        end
+
+        # @see #configure
+        # @see #configure_with_model
+        #
+        # @param config_json [Hash, nil] see Agama::Storage::Manager#configure
+        def calculate_proposal(config_json = nil)
+          backend.configure(config_json)
+          self.ProposalChanged(recover_proposal)
+          finish_progress
```

Comment:
I would extract the `finish_progress` call. It is weird to not see it in the methods calling to `#calculate_proposal`.